controllers/statistics.js

```diff
@@ -0,0 +1,168 @@
+import models from '../models';
+import timeWasted from '../helpers/readingTime';
+
+const { User, article, statistic } = models;
+
+const userAttrib = ['username', 'image', 'bio'];
+const articleAttrib = ['title', 'description', 'slug', 'tagList'];
+
+const statistics = {
+  createStats: async (req, res) => {
+    const { id } = req.user;
+    const { timeSpentReading } = req.query;
+    const convertToMins = timeSpentReading / 60;
+    const { slug } = req.query;
+    const totalReadTime = timeWasted.nonReadTime(convertToMins);
+    try {
+      const articles = await article.findOne({
+        where: { slug },
+        attributes: ['id', 'title', 'body', 'slug'],
+        include: {
+          model: User,
+          attributes: userAttrib
+        }
+      });
+      if (totalReadTime <= 0.3333) {
+        return res.status(409).send({
```

Comment:
I think that the response code used here is not correct! **409** means that there is a **conflict**. I think that we should use something like **403** that means **forbidden**. As we want to allow users who have read the article enough time.

Reply:
@avpaul basing on [this](https://stackoverflow.com/questions/45063805/distinguishing-http-status-code-403-and-409-in-practice-or-400). 
I will stay with **409**

Reply:
@avpaul after our 1:1 we have agreed that is a 400.